sfwebui.py

```diff
@@ -363,27 +362,35 @@ def rerunscan(self, id):
             scantarget = scantarget.lower()
 
         # Start running a new scan
-        newId = sf.genScanInstanceGUID(scanname)
-        p = mp.Process(target=SpiderFootScanner, args=(scanname, scantarget, targetType, newId,
-                modlist, cfg, modopts))
-        p.start()
+        try:
+            s = SpiderFootScanner(scanname, scantarget, targetType, modlist, cfg)
+            scanId = s.getId()
```

Comment:
Race condition mentioned above.

Reply:
Disagree. The ID is set during SpiderFootScanner initialisation. This line is called after the SpiderFootScanner object is created.